nilearn/datasets/tests/test_atlas.py

```diff
@@ -266,3 +266,19 @@ def test_fetch_atlas_yeo_2011():
     assert_true(isinstance(dataset.thin_17, _basestring))
     assert_true(isinstance(dataset.thin_7, _basestring))
     assert_equal(len(mock_url_request.urls), 1)
+
+
+@with_setup(setup_mock, teardown_mock)
+@with_setup(setup_tmpdata, teardown_tmpdata)
+def test_fetch_atlas_aal_spm_12():
+    ho_dir = os.path.join(tmpdir, 'aal_spm_12', 'aal_for_SPM12')
+    os.makedirs(ho_dir)
+    dummy = open(os.path.join(ho_dir, 'AAL.xml'), 'w')
```

Comment:
you can use a context manager here:

```python
with open(os.path.join(ho_dir, 'AAL.xml'), 'w') as xml_file:
    xml_file.write(...)
```

and the file will be closed at the exit of the block.